WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderCardUiState.kt

```diff
@@ -56,14 +60,42 @@ sealed class ReaderCardUiState {
             val imageType: ImageType,
             val onDiscoverClicked: ((Long, Long) -> Unit)
         )
-
-        data class ActionUiState(
-            val isEnabled: Boolean,
-            val isSelected: Boolean = false,
-            val contentDescription: UiString? = null,
-            val count: Int = 0,
-            val onClicked: ((Long, Long, Boolean) -> Unit)? = null
-        )
     }
 }
 
+sealed class ReaderPostCardAction {
+    abstract val type: ReaderPostCardActionType
+    open val onClicked: ((Long, Long, ReaderPostCardActionType) -> Unit)? = null
+    open val isSelected: Boolean = false
+
+    data class PrimaryAction(
+        val isEnabled: Boolean,
+        val contentDescription: UiString? = null,
+        val count: Int = 0,
+        override val isSelected: Boolean = false,
+        override val type: ReaderPostCardActionType,
+        override val onClicked: ((Long, Long, ReaderPostCardActionType) -> Unit)? = null
+    ) : ReaderPostCardAction()
+
+    data class SecondaryAction(
+        val label: UiString,
+        @AttrRes val labelColor: Int,
+        @DrawableRes val iconRes: Int,
+        @AttrRes val iconColor: Int = labelColor,
+        override val isSelected: Boolean = false,
+        override val type: ReaderPostCardActionType,
+        override val onClicked: (Long, Long, ReaderPostCardActionType) -> Unit
+    ) : ReaderPostCardAction()
+}
+
+enum class ReaderPostCardActionType {
```

Comment:
I'm not sure about this whole `ReaderPostCardAction` structure. I definitely plan to improve it a bit more. For example `ReaderPostCardActionType` contains both Primary buttons and the buttons hidden under the "more/..." menu, which might be a bit confusing. Any suggestions are welcome :).

Reply:
Agree, it looks bit odd to have `ReaderPostCardActionType` with both primary and secondary actions without any distinction. Can't it be converted to a nested sealed class and accessed like `ReaderPostCardActionType.Primary.Bookmark`?

Reply:
👍 Good idea, I'll do that in a new PR. ;)